src/catalog/src/durable/persist.rs

```diff
@@ -977,9 +977,7 @@ impl UnopenedPersistCatalogState {
         let since_handle = persist_client
             .open_critical_since(
                 catalog_shard_id,
-                // TODO: We may need to use a different critical reader
-                // id for this if we want to be able to introspect it via SQL.
-                PersistClient::CONTROLLER_CRITICAL_SINCE,
+                PersistClient::CATALOG_CRITICAL_SINCE,
```

Comment:
I actually think we just want to delete critical sinces from the durable catalog code. AFAICT the catalog only uses this to ensure that things keep compacted, which the storage controller will now handle. In particular, it doesn't seem to use the critical since as an additional read capability. Though definitely we should get a @jkosh44 double check on this.

Reply:
Yeah, I'm good with that if you prefer! I'm 99% sure that'll work just fine from an implementation perspective. From a philosophical perspective, I wasn't sure whether we liked the idea of the catalog's read policy being determined by the read policy on `mz_internal.mz_catalog_raw`, or whether we wanted to keep our own read capability around (i.e., what this PR does).

Reply:
> AFAICT the catalog only uses this to ensure that things keep compacted, which the storage controller will now handle. In particular, it doesn't seem to use the critical since as an additional read capability. Though definitely we should get a @jkosh44 double check on this.

This is correct, we're always reading the most recent timestamp. So we only use this to compact the shard and ensure that some timestamp is readable, i.e. to keep `since <= upper` unless there's some other mechanism that does this..

Reply:
Right, with this PR, the storage controller is opening a critical since handle that has a read policy of lagging the upper by 1. So if you're comfortable relying on the storage controller to manage the since handle, we can just remove the critical since handle in the `catalog` crate entirely.